### DISCUSSION TITLE
Question regarding Clients conforming to Sendable Protocol

### DISCUSSION THREAD

**Discussion by inwoodev** [pointfreeco/swift-composable-architecture#1534] [Q&A] [answered]
While going throuch TCA practice apps, I had a question in my mind regarding conformance to Sendable Protocol.

One could see that the `WeatherClient` in SearchApp does conform to Sendable Protocol:
``` swift
struct WeatherClient: Sendable {
  var forecast: @Sendable (GeocodingSearch.Result) async throws -> Forecast
  var search: @Sendable (String) async throws -> GeocodingSearch
}
```

However, other clients of sample apps do not conform to `Sendable`
``` swift
struct AudioPlayerClient {
  var play: @Sendable (URL) async throws -> Bool
}
```

``` swift
struct SpeechClient {
  var finishTask: @Sendable () async -> Void
  var requestAuthorization: @Sendable () async -> SFSpeechRecognizerAuthorizationStatus
  var startTask:
    @Sendable (SFSpeechAudioBufferRecognitionRequest) async -> AsyncThrowingStream<
      SpeechRecognitionResult, Error
    >

  enum Failure: Error, Equatable {
    case taskError
    case couldntStartAudioEngine
    case couldntConfigureAudioSession
  }
}
```

According to my understanding of Sendable, all structs are implicitly `Sendable` if it has sendable members only(maybe I misunderstood this part). If my understanding is correct, wouldn't it be alright to discard Sendable protocol from WeatherClient?:
``` swift
struct WeatherClient {
  var forecast: @Sendable (GeocodingSearch.Result) async throws -> Forecast
  var search: @Sendable (String) async throws -> GeocodingSearch
}
```
Since all the properties of WeatherClient are sendable, I thought that WeatherClient would implicitly be sendable as well and there would be no need of explicit conformance to `Sendable`.

If I am mistaken, and that there is a valid reason of differentiating `WeatherClient` from other clients, I would like to know why...

**Comment by tgrapperon** [comment, marked as answer]
Hey @inwoodev, you're right, this is mostly an oversight. It could probably be removed, and you're not missing anything special.
Implicit conformance to `Sendable` is not possible when the value type is generic, or of course when it hosts non `Sendable` properties. This is not the case here, and `WeatherClient` is implicitly `Sendable`.

**Reply by inwoodev** [reply]
Thank your! Everything is clear now!! :)

**Comment by tgrapperon** [comment]
In any case someone stumbles upon this discussion in the future, implicit sendability is only valid at the module's scope (that is `internal`). You may want to explicitly conform to `Sendable` if you intend to define a dependency in its own dedicated module for example, as @mbrandonw pointed in #1535.

**Reply by inwoodev** [reply]
Good point to include!!
